tls/s2n_ktls_parameters.h

```diff
@@ -24,21 +24,66 @@
  * - https://elixir.bootlin.com/linux/v6.3.8/A/ident/SOL_TCP
  */
 
-#define S2N_TLS_ULP_NAME      "tls"
-#define S2N_TLS_ULP_NAME_SIZE sizeof(S2N_TLS_ULP_NAME)
-
 #if defined(__linux__)
     #define S2N_KTLS_SUPPORTED true
 
     /* socket definitions */
     #define S2N_TCP_ULP 31 /* Attach a ULP to a TCP connection.  */
     #define S2N_SOL_TCP 6  /* TCP level */
+    #define S2N_SOL_TLS 282
+    #define S2N_TLS_TX  1 /* Set transmit parameters */
+    #define S2N_TLS_RX  2 /* Set receive parameters */
+
+    /* cipher definitions */
+    #define S2N_TLS_CIPHER_AES_GCM_128              51
+    #define S2N_TLS_CIPHER_AES_GCM_128_IV_SIZE      8
+    #define S2N_TLS_CIPHER_AES_GCM_128_KEY_SIZE     16
+    #define S2N_TLS_CIPHER_AES_GCM_128_SALT_SIZE    4
+    #define S2N_TLS_CIPHER_AES_GCM_128_TAG_SIZE     16
+    #define S2N_TLS_CIPHER_AES_GCM_128_REC_SEQ_SIZE 8
 
 #else
-    /* For unsupported platforms 0-init all values. */
-    #define S2N_KTLS_SUPPORTED false
+    /* For unsupported platforms 0-init (array of size 1) all values. */
+    #define S2N_KTLS_SUPPORTED                      false
 
     /* socket definitions */
-    #define S2N_TCP_ULP        0
-    #define S2N_SOL_TCP        0
+    #define S2N_TCP_ULP                             0
+    #define S2N_SOL_TCP                             0
+    #define S2N_SOL_TLS                             0
+    #define S2N_TLS_TX                              0
+    #define S2N_TLS_RX                              0
+
+    /* cipher definitions */
+    #define S2N_TLS_CIPHER_AES_GCM_128              0
+    #define S2N_TLS_CIPHER_AES_GCM_128_IV_SIZE      1
+    #define S2N_TLS_CIPHER_AES_GCM_128_KEY_SIZE     1
+    #define S2N_TLS_CIPHER_AES_GCM_128_SALT_SIZE    1
+    #define S2N_TLS_CIPHER_AES_GCM_128_TAG_SIZE     1
+    #define S2N_TLS_CIPHER_AES_GCM_128_REC_SEQ_SIZE 1
+
 #endif
+
+/* Common */
+#define S2N_TLS_ULP_NAME      "tls"
+#define S2N_TLS_ULP_NAME_SIZE sizeof(S2N_TLS_ULP_NAME)
+
+typedef unsigned short __u16;
+
+struct s2n_tls_crypto_info {
+    __u16 version;
+    __u16 cipher_type;
+};
```

Comment:
If this format is required by setsockopt, why are we defining it here? Minimally this needs a comment + shouldn't be named like it's an internal s2n-tls thing.